include/libtorrent/aux_/session_impl.hpp

```diff
@@ -117,8 +118,14 @@ namespace dht {
 		class item;
 	}
 
-	struct listen_socket_t
+	struct listen_socket_t final : aux::session_listen_socket
```

Comment:
I just realized this class is defined directly in the ``libtorrent`` namespace, not under ``aux``. It doesn't look like this changed anytime recently. I will attempt to move it into ``aux`` where I think it belongs